xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-ui/src/main/resources/Main/SolrSearchMacros.xml

```diff
@@ -67,8 +67,11 @@
     &lt;/div&gt;
     &lt;div class="col-xs-12 col-sm-6"&gt;
       &lt;div class="input-group"&gt;
-        &lt;input type="search" name="text" class="form-control withTip useTitleAsTip"
+        &lt;input id='search-page-bar-input' type="search" name="text" class="form-control withTip useTitleAsTip"
```

Comment:
Do we have any policy how to name such ids? Also, do we really need an additional label? From my understanding of https://www.w3.org/TR/WCAG20-TECHS/H65.html this shouldn't be needed.

Reply:
> Also, do we really need an additional label?

It's not "needed", but [recommended by WAI ARIA](https://www.w3.org/WAI/ARIA/apg/practices/names-and-descriptions/#accessiblenameguidancebyrole) to get label for those elements.

And also recommended by HTML:
> Relying on the [title](https://html.spec.whatwg.org/multipage/dom.html#attr-title) attribute is currently discouraged as many user agents do not expose the attribute in an accessible manner as required by this specification (e.g., requiring a pointing device such as a mouse to cause a tooltip to appear, which excludes keyboard-only users and touch-only users, such as anyone with a modern phone or tablet).

From https://html.spec.whatwg.org/multipage/dom.html#the-title-attribute


Note that the title attribute is nice, but some AT don't support it properly, so the rule of thumb I kept is: have at least one kind of 'real' label (label, aria-labelledby, aria-label) and if you don't mind the hover popup, add a title with the same content.
A couple of sources to support this: 
* https://adrianroselli.com/2020/01/my-priority-of-methods-for-labeling-a-control.html
* https://www.tpgi.com/using-the-html-title-attribute/

Reply:
> Do we have any policy how to name such ids?

I just figured such an id would be okay since it's quite specific and found only once on the page when testing. I don't know of any name policy for ids.

Reply:
> Note that the title attribute is nice, but some AT don't support it properly, so the rule of thumb I kept is: have at least one kind of 'real' label (label, aria-labelledby, aria-label) and if you don't mind the hover popup, add a title with the same content. A couple of sources to support this:
> 
> * https://adrianroselli.com/2020/01/my-priority-of-methods-for-labeling-a-control.html
> * https://www.tpgi.com/using-the-html-title-attribute/

At least the second resource doesn't support this. The hidden label element has the exactly same accessibility problems, actually, additionally, it cannot even be accessed with a pointing device, so the accessibility is even worse. From the second link:

> For a label on a control that has no visible text label: `<input type="text" title="name">`
> Screen readers users will have access to the control label, as the title attribute is mapped to the accessible name property in accessibility APIs (when a text label using the label element is not supplied ). Many other users will not. Recommend including a visible text label whenever possible.

So it explicitly states that screen reader users have access to the title. What you're proposing here isn't better in any way from what I understand as it is even less visible than the title attribute.

Reply:
Addressed in https://github.com/xwiki/xwiki-platform/pull/2187/commits/7c4005a71dc38a98189bd6d511099567b7bbc96a
See https://github.com/xwiki/xwiki-platform/pull/2187#discussion_r1235402944